aclmapping/evm/mappings.go

```diff
@@ -0,0 +1,95 @@
+	acltypes "github.com/cosmos/cosmos-sdk/x/accesscontrol/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	evmkeeper "github.com/sei-protocol/sei-chain/x/evm/keeper"
+	evmtypes "github.com/sei-protocol/sei-chain/x/evm/types"
+)
+
+var ErrInvalidMessageType = fmt.Errorf("invalid message received for EVM Module")
+
+func GetEVMDependencyGenerators(evmKeeper evmkeeper.Keeper) aclkeeper.DependencyGeneratorMap {
+	dependencyGeneratorMap := make(aclkeeper.DependencyGeneratorMap)
+	EVMTransactionMsgKey := acltypes.GenerateMessageKey(&evmtypes.MsgEVMTransaction{})
+	dependencyGeneratorMap[EVMTransactionMsgKey] = func(k aclkeeper.Keeper, ctx sdk.Context, msg sdk.Msg) ([]sdkacltypes.AccessOperation, error) {
+		return TransactionDependencyGenerator(k, evmKeeper, ctx, msg)
+	}
+
+	return dependencyGeneratorMap
+}
+
+func TransactionDependencyGenerator(_ aclkeeper.Keeper, evmKeeper evmkeeper.Keeper, ctx sdk.Context, msg sdk.Msg) ([]sdkacltypes.AccessOperation, error) {
+	evmMsg, ok := msg.(*evmtypes.MsgEVMTransaction)
+	if !ok {
+		return []sdkacltypes.AccessOperation{}, ErrInvalidMessageType
+	}
+
+	tx, _ := evmMsg.AsTransaction()
+	// Only specifying accesses to `to` address since `from` has to be derived via signature,
+	// which happens in the ante handler (i.e. after this generator is called) and is quite heavy
+	// so we don't want to repeat it.
+	toAddress := tx.To()
+	seiAddress, associated := evmKeeper.GetSeiAddress(ctx, *toAddress)
+	var idTempl string
+	var resourceType sdkacltypes.ResourceType
+	if associated {
+		idTempl = hex.EncodeToString(banktypes.CreateAccountBalancesPrefix(seiAddress))
+		resourceType = sdkacltypes.ResourceType_KV_BANK_BALANCES
+	} else {
+		idTempl = hex.EncodeToString(evmtypes.BalanceKey(*toAddress))
+		resourceType = sdkacltypes.ResourceType_KV_EVM_BALANCE
+	}
+
+	return []sdkacltypes.AccessOperation{
+		{
+			AccessType:         sdkacltypes.AccessType_READ,
+			ResourceType:       resourceType,
+			IdentifierTemplate: idTempl,
+		},
+		{
+			AccessType:         sdkacltypes.AccessType_WRITE,
+			ResourceType:       resourceType,
+			IdentifierTemplate: idTempl,
+		},
+		{
+			AccessType:         sdkacltypes.AccessType_READ,
+			ResourceType:       sdkacltypes.ResourceType_KV_EVM,
+			IdentifierTemplate: "*", // no way to derive what fields a contract might read
+		},
+		{
+			AccessType:         sdkacltypes.AccessType_WRITE,
+			ResourceType:       sdkacltypes.ResourceType_KV_EVM,
+			IdentifierTemplate: "*", // no way to derive what fields a contract might write
+		},
+		{
+			AccessType:         sdkacltypes.AccessType_READ,
+			ResourceType:       sdkacltypes.ResourceType_KV_BANK,
+			IdentifierTemplate: "*", // from address access (reasoning described above)
+		},
+		{
+			AccessType:         sdkacltypes.AccessType_WRITE,
+			ResourceType:       sdkacltypes.ResourceType_KV_BANK,
+			IdentifierTemplate: "*", // from address access (reasoning described above)
```

Comment:
ideally we'd want to know a sepcific bank balance key for the "from" field as well, is there another way to get the from addr from the derived signature? it think for now its fine if we exclude it, but would be nice to have down the road

Reply:
@udpatil it should be possible to frontload stateless ante operations like signature verification (which would give us the from address) to happen before ACL is evaluated, but that'd be a major refactor

Reply:
or, we allow `CheckTx` to modify tx bytes by settings fields that are derived during CheckTx ante run. Not sure if that'd have any security implication or not

Reply:
yeah i think its fine to leave as is for now, maybe in the future when we look to add the pre cached dependencies from checkTX or something we can bundle this in there